src/libs/actions/IOU.js

```diff
@@ -943,21 +962,39 @@ function getSendMoneyParams(report, amount, currency, comment, paymentMethodType
             key: optimisticChatReportData.key,
             value: {pendingFields: null},
         });
+        failureData.push(
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: optimisticChatReportData.key,
+                value: {
+                    errorFields: {
+                        createChat: ErrorUtils.getMicroSecondOnyxError('report.genericCreateReportFailureMessage'),
+                    },
+                },
+            },
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticIOUReport.reportID}`,
+                value: {
+                    [optimisticIOUReportAction.reportActionID]: {
+                        errors: ErrorUtils.getMicroSecondOnyxError(null),
+                    },
+                },
+            },
+        );
+
+        // Add an optimistic created action to the optimistic reportActions data
+        optimisticReportActionsData.value[optimisticCreatedAction.reportActionID] = optimisticCreatedAction;
+    } else {
         failureData.push({
             onyxMethod: Onyx.METHOD.MERGE,
-            key: optimisticChatReportData.key,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticIOUReport.reportID}`,
             value: {
-                errorFields: {
-                    createChat: ErrorUtils.getMicroSecondOnyxError('report.genericCreateReportFailureMessage'),
+                [optimisticIOUReportAction.reportActionID]: {
+                    errors: ErrorUtils.getMicroSecondOnyxError('iou.error.other'),
                 },
             },
         });
```

Comment:
I don't see why we are conditionally setting the error of the IOU report action based on the CHAT report.
Chat reports and IOU reports are not rendered together, so why we are making this change?

Reply:
I'm actually not sure about this one, I'll have to look at the code a bit more to see what's going on but this was the original code in `main`, I didn't add anything that was not necessary.

https://github.com/Expensify/App/blob/14e3e1b848ac7df4505d87685552cff8f7c7833f/src/libs/actions/IOU.js#L959-L960

This code was written by @yuwenmemon so maybe he can shed some light on this?

Reply:
I added this so we could clear out all failure data for IOU, because the error for creating the report itself should supercede it here. You can check out the PR here: https://github.com/Expensify/App/pull/18328

Whether it was the correct way to go... 🤷‍♂️

Reply:
@s77rt it looks like this is for the iou report action in chats, that should answer your question.

Reply:
I haven't tested this case as I didn't have the steps before (in the PR linked), if possible can you test for it while creating the reviewer checklist since it's already day 2? I'll be able to test it only tomorrow morning (IST).

Reply:
I guess we can move on then, just please remove the redundant `optimisticIOUReportAction.reportActionID` `failureData` set above the if condition (it will be overwritten).

https://github.com/Expensify/App/blob/999cd0bb7052929a70db2cdb6d3b0386c39180e4/src/libs/actions/IOU.js#L932-L940

Reply:
Ah I forgot this. Done!